_studio/mfx_lib/genx/h264_encode/src/genx_histogram.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2015-2018 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#if defined(_WIN32) || defined(_WIN64)
+
+#include "cm/cm.h"
+
+static const ushort usSeq0_15[16] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+static const ushort uiSeq0_7[8] = { 0, 1, 2, 3, 4, 5, 6, 7 };
+
+extern "C" _GENX_MAIN_ void HistogramSLMFrame(
+    SurfaceIndex INBUF,
+    SurfaceIndex OUTBUF,
+    uint max_h_pos,
+    uint max_v_pos,
+    uint off_h,
+    uint off_v)
+{
+    const uint SLM_SIZE_INT = 256;
+    const uint SLM_SIZE = SLM_SIZE_INT * sizeof(uint);
+
+    vector<uint, 64> vUI64;
```

Comment:
Would be nice to have includes for vector, matrix etc. here

Reply:
They are in MDF package for compiler, you are not including std into your code right?

Reply:
@lakulako, they are defined at <cm_vm.h> that is included by <cm/cm.h>